pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/HiveORCAccessor.java

```diff
@@ -0,0 +1,152 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import org.apache.hadoop.hive.ql.io.orc.OrcInputFormat;
+import org.apache.hadoop.hive.ql.io.sarg.SearchArgument;
+import org.apache.hadoop.hive.ql.io.sarg.SearchArgumentFactory;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.Reporter;
+import org.apache.hawq.pxf.api.FilterParser;
+import org.apache.hawq.pxf.api.utilities.ColumnDescriptor;
+import org.apache.hawq.pxf.api.utilities.InputData;
+import org.apache.commons.lang.StringUtils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.hawq.pxf.plugins.hive.HiveInputFormatFragmenter.PXF_HIVE_SERDES;
+
+/**
+ * Specialization of HiveAccessor for a Hive table that stores only ORC files.
+ * This class replaces the generic HiveAccessor for a case where a table is stored entirely as ORC files.
+ * Use together with {@link HiveInputFormatFragmenter}/{@link HiveColumnarSerdeResolver}
+ */
+public class HiveORCAccessor extends HiveAccessor {
+
+    private final String READ_COLUMN_IDS_CONF_STR = "hive.io.file.readcolumn.ids";
+    private final String READ_ALL_COLUMNS = "hive.io.file.read.all.columns";
+    private final String READ_COLUMN_NAMES_CONF_STR = "hive.io.file.readcolumn.names";
+    private final String SARG_PUSHDOWN = "sarg.pushdown";
+
+    /**
+     * Constructs a HiveORCFileAccessor.
+     *
+     * @param input input containing user data
+     * @throws Exception if user data was wrong
+     */
+    public HiveORCAccessor(InputData input) throws Exception {
+        super(input, new OrcInputFormat());
+        String[] toks = HiveInputFormatFragmenter.parseToks(input, PXF_HIVE_SERDES.COLUMNAR_SERDE.name(), PXF_HIVE_SERDES.LAZY_BINARY_COLUMNAR_SERDE.name(), PXF_HIVE_SERDES.ORC_SERDE.name(), PXF_HIVE_SERDES.VECTORIZED_ORC_SERDE.name());
+        initPartitionFields(toks[HiveInputFormatFragmenter.TOK_KEYS]);
+        filterInFragmenter = new Boolean(toks[HiveInputFormatFragmenter.TOK_FILTER_DONE]);
+    }
+
+    @Override
+    public boolean openForRead() throws Exception {
+        addColumns();
+        addFilters();
+        return super.openForRead();
+    }
+
+    /**
+     * Adds the table tuple description to JobConf ojbect
+     * so only these columns will be returned.
+     */
+    private void addColumns() throws Exception {
+
+        List<Integer> colIds = new ArrayList<Integer>();
+        List<String> colNames = new ArrayList<String>();
+        for(ColumnDescriptor col: inputData.getTupleDescription()) {
+            if(col.isProjected()) {
+                colIds.add(col.columnIndex());
+                colNames.add(col.columnName());
+            }
+        }
+        jobConf.set(READ_ALL_COLUMNS, "false");
```

Comment:
I didn't want to complicate this function for such a specific case. Most common case is for query with variable number of columns